### PR TITLE
Implement changes to accommodate to GUI

### PR DESCRIPTION
**Pull request by dannytayjy**
- Make changes to MarkTask, UnmarkTask, DeleteEmployee, DeleteTask, TagTask, UntagTask and TagPriority to ensure that GUI view is updated

### CODE REVIEW

File: src/main/java/manageezpz/commons/core/Messages.java [hunk 1]
```diff
@@ -5,17 +5,20 @@
...
-    public static final String MESSAGE_INVALID_TASK_DISPLAYED_INDEX = "The task index provided is invalid";
+    public static final String MESSAGE_EMPTY_NAME = "Name field cannot be empty! \n%1$s";
+    public static final String MESSAGE_EMPTY_PRIORITY = "Priority field cannot be empty! \n%1$s";
+    public static final String MESSAGE_INVALID_PRIORITY =
+            "Invalid priority! " + "Valid priorities are NONE/LOW/MEDIUM/HIGH. \n%1$s";
+    public static final String MESSAGE_EMPTY_TASK_NUMBER = "Task number field cannot be empty! \n%1$s";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX =
+            "The person index provided is invalid as it exceeds the amount of tasks in the displayed list! \n%1$s";
+    public static final String MESSAGE_INVALID_TASK_DISPLAYED_INDEX =
+            "The task index provided is invalid as it exceeds the amount of tasks in the displayed list! \n%1$s";
```
Comment (Denniszedead) [inline src/main/java/manageezpz/commons/core/Messages.java:18]: 👍🏻  Good job in adding the various errors

File: src/main/java/manageezpz/logic/commands/TagTaskCommand.java [hunk 2]
```diff
@@ -5,67 +5,80 @@
...
-            + "1 "
-            + PREFIX_NAME + "Alex Yeoh";
-    public static final String MESSAGE_SUCCESS = "Task has been tagged! : %1$s";
-    public static final String MESSAGE_DUPLICATE_TASK = "This Task has already been assigned to the same person!";
+            + PREFIX_NAME + "EMPLOYEE_NAME\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_NAME + "Alex Yeoh";
+
+    public static final String MESSAGE_TAG_TASK_SUCCESS = "Employee %1$s is tagged to the task: ";
+
+    public static final String MESSAGE_NO_SUCH_PERSON = "There is no employee with the name %1$s!";
```
Comment (Denniszedead) [inline src/main/java/manageezpz/logic/commands/TagTaskCommand.java:27]: 👍🏻  Good that you indicate on the error messages for the tag task command

File: src/main/java/manageezpz/model/task/UniqueTaskList.java [hunk 3]
```diff
@@ -57,28 +59,90 @@ public void setTask(Task target, Task editedTask) {
...
+        if (task instanceof Todo) {
+            return new Todo((Todo) task);
+        } else if (task instanceof Deadline) {
+            return new Deadline((Deadline) task);
+        } else if (task instanceof Event) {
+            return new Event((Event) task);
+        } else {
+            // The else statement should not be reached since there are
+            // only three types of tasks, i.e., todo, deadline and event
+            throw new InvalidTaskTypeException();
```
Comment (Denniszedead) [inline src/main/java/manageezpz/model/task/UniqueTaskList.java:145]: 👍🏻  Good use of defensive programming where if the task do not follow the three types of tasks then it shows an error. However, for such situation, assertions will be nice.
Try: `assert false : "Invalid task type"`

**Review by Denniszedead — APPROVED** [review]

File: src/main/java/manageezpz/model/task/UniqueTaskList.java (hunk 3 above)
Comment (dannytayjy) [reply src/main/java/manageezpz/model/task/UniqueTaskList.java:145]: Great catch, we shall change it in v1.4!

**Pull request merged**